java/src/jmri/jmrit/display/layoutEditor/LayoutTrack.java

```diff
@@ -293,6 +293,33 @@ public String getTurnoutStateString(int turnoutState) {
         return result;
     }
 
+    /**
+     * Check for active block boundaries.
+     * <p>
+     * If any connection point of a track object has attached objects, such as signal
+     * masts, signal heads or NX sensors, the track component cannot be deleted.
+     * @return true if the track component can be deleted.
+     */
+    public abstract boolean canRemoveTrackComponent();
```

Comment:
Isn’t the “TrackComponent” in the name redundant since this is a question asked of the LayoutTrack object?

Reply:
I have changed the method name to canRemoveObject.  I have trouble with just canRemove since it feels incomplete.

Reply:
I personally would prefer just canRemove().  I still see the Object as redundant.  Remember this is always going to be called as called as foo.canRemoveObject() where foo is an Object.

@rhwood @bobjacobsen do you have a preference on the method name here?